### PR TITLE
Update koa-helmet to be a simple `helmet` wrapper

### PR DESCRIPTION
**Pull request by venables**
In an effort to make `helmet` updates available sooner (#12, #13, https://github.com/helmetjs/helmet/issues/100), this is a first draft of updating `koa-helmet` to be a simple wrapper around `helmet` - with only one dependency: helmet.

### CODE REVIEW

**Comment by EvanHahn** [comment]
I'm not an expert on Koa, but this looks good to me. I'd like to see a few more tests before putting this into the `helmetjs` GitHub org, but this is great.

Love seeing a diff that deletes 1,438 lines!

**Comment by venables** [comment]
>  I'd like to see a few more tests

As would I - I'll add one for the defaults case, which will bring it up to par with the helmetjs/helmet repo's test coverage. 

And, happy to leave this as venables/koa-helmet at first if you're concerned

**Comment by EvanHahn** [comment]
Sure! I don't feel strongly about putting this under the `helmetjs` org. I'll definitely recommend this module from the Helmet readme once this is done, though.

File: lib/index.js [hunk 1]
```diff
@@ -1,34 +1,21 @@
...
-  var middleware;
-  var name;
-
-  if (/\.js$/.test(filename)) {
-    name = filename.substr(0, filename.lastIndexOf('.'));
-    middleware = require('./middleware/' + name);
-
-    module.exports[name] = middleware;
+var koaHelmet = function() {
+  return function * (next) {
```
Comment (MarkHerhold) [inline lib/index.js]: `function * (next)` is a little spaced out.

File: lib/index.js [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
-  overrides.defaults = false;
+};
 
-  for (var name in module.exports) {
-    if (overrides[name] !== false) {
-      modules.push(module.exports[name]());
+Object.keys(helmet).forEach(function(helmetMethod) {
+  koaHelmet[helmetMethod] = function() {
+    return function *(next) {
+      yield helmet[helmetMethod]().bind(null, this.req, this.res);
```
Comment (MarkHerhold) [inline lib/index.js]: Loss of passed in arguments. Please see PR comments.

**Comment by MarkHerhold** [comment]
A few things:

1. Thank you both for collaborating on this. This is why open source is awesome. :+1: 
1. I'd rather see `koa-helmet` become a member of the `helmetjs` org. I think it looks more professional that way and there are fewer questions about quality and trust. This is up to you two though, so I'll let it go.
1. The current PR doesn't work with options as far as I can tell. I need to do more testing but I think the code should look more like this:

Koa App:
```js
var app = require('koa')();
var helmet = require('koa-helmet');
// let's say I only want to add a fake X-Powered-By header
app.use(helmet.hidePoweredBy({ setTo: 'PHP 4.2.0' }));
```

Test:
```bash
curl localhost:3000 -I
    HTTP/1.1 404 Not Found
    X-Powered-By: PHP 4.2.0 <-- not present !
    Content-Type: application/json; charset=utf-8
    ...
```

Code that needs to be (index.js):
```js
Object.keys(helmet).forEach(function(helmetMethod) {
    koaHelmet[helmetMethod] = function(options) {
        console.log('options: ' + JSON.stringify(options)); // <-- take a look at the options here :)
        return function *(next) {
            yield helmet[helmetMethod](options).bind(null, this.req, this.res);
            yield next;
        };
    };
});
```

**Comment by venables** [comment]
@MarkHerhold passing arguments now.

Also going to follow semver.  It would be helpful if helmet did the same so we can properly use the caret option to prevent breaking changes.  E.g. all helmet wrappers in version 1.x.x will have no breaking changes when helmet updates code within 1.x.x

**Comment by EvanHahn** [comment]
@venables Yes. We're [on the road to releasing 1.0.0](https://github.com/helmetjs/helmet/issues/92) and the main holdups are in the CSP middleware.

For what it's worth, none of the middlewares will be changed until 1.0.0, where one deprecated one will be removed. Until then, you could probably pin to `0.x`.

**Comment by MarkHerhold** [comment]
@venables You may want to update the license too

**Pull request merged**